### PR TITLE
chore(deploy): add deploy workflow

### PR DESCRIPTION
**Pull request by Cali93**
This probably needs more work and configuration.

Who could take it from here or help me during a pair programming session ?

### CODE REVIEW

**Review by xav — APPROVED** [review]
LGTM

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      - name: Initialize Terraform
+        run: terraform init
+
+      - name: Log in to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+        with:
+          region: ${{ env.AWS_REGION }}
+
+      - name: Build and push Docker image to Amazon ECR
```
Comment (HarryET) [inline .github/workflows/cd.yaml:44]: You can use docker build here, have a look at Echo's deploy here: https://github.com/WalletConnect/echo-server/blob/85b8c485a4645ee231de8662b6875809ab775837/.github/workflows/release.yml#L110-L137

File: .github/workflows/cd.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          docker tag ${env.ECR_REPOSITORY_NAME}:${{ github.sha }} ${env.ACCOUNT_ID}.dkr.ecr.${env.AWS_REGION}.amazonaws.com/${env.ECR_REPOSITORY_NAME}:${{ github.sha }}
+          docker push ${env.ACCOUNT_ID}.dkr.ecr.${env.AWS_REGION}.amazonaws.com/${env.ECR_REPOSITORY_NAME}:${{ github.sha }}
+
+      - name: Deploy to Amazon ECS
+        uses: hashicorp/terraform-github-actions@master
+        with:
+          tf_actions_version: "0.2.0"
+          working-directory: ./terraform
+          terraform-actions-command: "apply"
+          args: |
```
Comment (HarryET) [inline .github/workflows/cd.yaml:56]: You can use env vars for the Terraform variables, https://developer.hashicorp.com/terraform/cli/config/environment-variables#tf_var_name

File: .github/workflows/cd.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
+name: Deploy to ECS
+
+on:
+  push:
+    branches:
+      - main
+
+env:
```
Comment (HarryET) [inline .github/workflows/cd.yaml:8]: If these are only used once why not inline them? Also the cluster name and service name, etc aren't needed terraform manages them

File: .github/workflows/cd.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  ECS_CLUSTER_NAME: cloud-auth
+  ECS_SERVICE_NAME: cloud-auth-service
+  TASK_FAMILY_NAME: cloud-auth
+  PORT: 3333
+  CLOUD_APP_ORIGIN: ${{ secrets.CLOUD_APP_ORIGIN }}
+  SUPABASE_JWT_SECRET: ${{ secrets.SUPABASE_JWT_SECRET }}
+  COOKIE_NAME: ${{ secrets.COOKIE_NAME }}
+  COOKIE_SECRET: ${{ secrets.COOKIE_SECRET }}
+  DIRECT_URL: ${{ secrets.DIRECT_URL }}
+  DATABASE_URL: ${{ secrets.DATABASE_URL }}
```
Comment (HarryET) [inline .github/workflows/cd.yaml:20]:
```suggestion
env:
  AWS_REGION: eu-central-1
  CLOUD_APP_ORIGIN: ${{ secrets.CLOUD_APP_ORIGIN }}
  SUPABASE_JWT_SECRET: ${{ secrets.SUPABASE_JWT_SECRET }}
  COOKIE_NAME: ${{ secrets.COOKIE_NAME }}
  COOKIE_SECRET: ${{ secrets.COOKIE_SECRET }}
  DIRECT_URL: ${{ secrets.DIRECT_URL }}
  DATABASE_URL: ${{ secrets.DATABASE_URL }}
```

File: .github/workflows/cd.yaml (hunk 3 above)
Comment (HarryET) [reply .github/workflows/cd.yaml:8]: Also a couple question what is, `DIRECT_URL` and `DATABASE_URL` refers to the hosted supabase db right?

**Review by HarryET — CHANGES_REQUESTED** [review]

**Pull request merged**